### PR TITLE
Add pygacode recipe

### PR DESCRIPTION
**Pull request by kalling**
Adding pygacode package

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by kalling** [comment]
Comment confirming I'm willing to be listed as a maintainer.

File: recipes/pygacode/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win or not x86_64]
+
+requirements:
+  build:
+    - python
+    - {{ compiler('fortran') }}
+    - numpy
```
Comment (scopatz) [inline recipes/pygacode/meta.yaml]: numpy and Python should not be build dependencies if they are linked against

File: recipes/pygacode/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  host:
+    - pip
+    - python
+    - numpy
+  run:
+    - python
+    - numpy
```
Comment (scopatz) [inline recipes/pygacode/meta.yaml]: See instructions for numpy at https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

File: recipes/pygacode/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    - openmp  # [linux]
```
Comment (scopatz) [inline recipes/pygacode/meta.yaml]: See OpenMP instructions at https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp

Comment (kalling) [reply recipes/pygacode/meta.yaml]: Thank you for the comments, they were quite helpful. Unfortunately, using libgomp as per instructions still results in `undefined symbol: omp_get_thread_num` during runtime.  Supposedly I should include `-fopenmp` when building, but I am not sure how to pass that to the numpy extension builder.  Do you happen to have a pointer for that?

Comment (scopatz) [reply recipes/pygacode/meta.yaml]: CC @isuruf

File: recipes/pygacode/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [py27 or win or not x86_64]
+
+requirements:
+  build:
+    - {{ compiler('fortran') }}
+    - libgomp      # [linux]
+    - llvm-openmp  # [osx]
```
Comment (scopatz) [inline recipes/pygacode/meta.yaml]: These should not be in build. Only in host.

Comment (isuruf) [reply recipes/pygacode/meta.yaml]: `libgomp` and `llvm-openmp` are special and should be in `build`.

Comment (scopatz) [reply recipes/pygacode/meta.yaml]: Oh weird. OK

File: recipes/pygacode/meta.yaml (hunk 3 above)
Comment (kalling) [reply recipes/pygacode/meta.yaml]: @scopatz I was able to resolve this by patching the `setup.py` file to make sure to link `libgomp`

**Pull request merged**